src/Microsoft.OData.Edm/Vocabularies/Annotations/EdmTermExtensions.cs

```diff
@@ -0,0 +1,446 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="EdmTermExtensions.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using Microsoft.OData.Edm.Csdl;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.Text;
+using Microsoft.OData.Edm.Csdl.Parsing.Ast;
+
+namespace Microsoft.OData.Edm.Vocabularies
+{
+    internal static class EdmTermExtensions
+    {
+        /// <summary>
+        /// Parses a <paramref name="defaultValue"/> into an IEdmExpression value of the correct EDM type.
+        /// </summary>
+        /// <param name=“typeReference”>The type of value.</param>
+        /// <param name=“defaultValue”>Original default value.</param>
+        /// <returns>An IEdmExpression of type <paramref name="typeReference".</paramref></returns>
+        public static IEdmExpression BuildDefaultEdmExpression(IEdmTypeReference typeReference, string defaultValue)
+        {
+            EdmTypeKind termTypeKind = typeReference.TypeKind();
+
+            // Create expressions/constants for the corresponding types
+            switch (termTypeKind) // TODO: Create test cases for all of these
+            {
+                case EdmTypeKind.Primitive:
+                    return BuildEdmPrimitiveValueExp(typeReference.AsPrimitive(), defaultValue);
+                case EdmTypeKind.Enum:
+                    // return BuildEdmEnumExp(typeReference.AsEnum(), defaultValue)
+                case EdmTypeKind.Complex:
+                case EdmTypeKind.Entity:
+                    // return BuildEdmStructuredTypeExp(typeReference.AsStructured(), defaultValue);
+                case EdmTypeKind.Collection:
+                    // return BuildEdmCollectionExp(typeReference, defaultValue);
+                case EdmTypeKind.Path:
+                    // return BuildEdmPathExp(typeReference.AsPath(), defaultValue);
+                case EdmTypeKind.TypeDefinition:
+                    // return BuildEdmTypeDefinitionExp(typeReference.AsTypeDefinition(), defaultValue);

```

Comment:
you have implementations for some of these. Which work and which don't? Do enum and typedefinition work?

Reply:
I hadn't tested the non-primitive types thoroughly, so that's why it's commented at the moment. I can see if I can test it once I've implemented the other feedback!

Reply:
[Deprioritizing for this PR]